dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/RumFileStrategy.kt

```diff
@@ -0,0 +1,33 @@
+package com.datadog.android.rum.internal.domain
+
+import android.content.Context
+import com.datadog.android.core.internal.domain.AsyncWriterFilePersistenceStrategy
+import com.datadog.android.core.internal.domain.PayloadDecoration
+import java.io.File
+
+internal class RumFileStrategy(
+    context: Context,
+    recentDelayMs: Long = MAX_DELAY_BETWEEN_LOGS_MS,
+    maxBatchSize: Long = MAX_BATCH_SIZE,
+    maxLogPerBatch: Int = MAX_ITEMS_PER_BATCH,
+    oldFileThreshold: Long = OLD_FILE_THRESHOLD,
+    maxDiskSpace: Long = MAX_DISK_SPACE
+) : AsyncWriterFilePersistenceStrategy<RumEvent>(
+    File(context.filesDir, RUM_FOLDER),
+    RumEventSerializer(),
+    recentDelayMs,
+    maxBatchSize,
+    maxLogPerBatch,
+    oldFileThreshold,
+    maxDiskSpace,
+    PayloadDecoration.NEW_LINE_DECORATION,
+    WRITER_THREAD_NAME
+) {
+    companion object {
+        internal const val TRACES_DATA_VERSION = 1
+        internal const val DATA_FOLDER_ROOT = "dd-rum"
+        internal const val RUM_FOLDER = "$DATA_FOLDER_ROOT-v$TRACES_DATA_VERSION"
+        internal const val MAX_DELAY_BETWEEN_LOGS_MS = 5000L
+        internal const val WRITER_THREAD_NAME = "ddog_rum_writer"
```

Comment:
For each feature (Logs / Tracing / RUM) we fork a new thread for writes and uploads. As threads on mobile are expensive and we can easily explode the number of threads by additional 6 (when all features are enabled), wouldn't it be beneficial at some point to reconsider this threading model? For example, do you think all features (Logs / Tracing / RUM) could use a single thread for writes? Or maybe this could be a user setting to choose expensive but performant multi-threading strategy in some apps and stay with limited threads count in others.

Reply:
I think there's no problem in using one Thread for all as they are each using a different folder. I think this is doable. Do you guys think we should add a task for this already ?

Reply:
I am actually thinking to use a ThreadPoolExecutor with say 2 threads and a BlockingQueue. Think this will work better there.

Reply:
I'm ok to have a ThreadPool indeed for the writes / reads. I added task 254 for that 👍